javaslang/src/main/java/javaslang/collection/LinkedHashSet.java

```diff
@@ -846,7 +852,13 @@ public String mkString(CharSequence prefix, CharSequence delimiter, CharSequence
 
     @Override
     public LinkedHashSet<Tuple2<T, Long>> zipWithIndex() {
-        return LinkedHashSet.ofAll(iterator().zipWithIndex());
+        return zipWithIndex(Tuple::of);
+    }
+
+    @Override
+    public <U> LinkedHashSet<U> zipWithIndex(BiFunction<? super T, ? super Long, ? extends U> mapper) {
```

Comment:
I've thought about it - and took a look at the Scala API reference. For example Scala's [TreeSet.zipWithIndex](http://www.scala-lang.org/api/2.11.8/index.html#scala.collection.immutable.TreeSet@zipWithIndex:scala.collection.immutable.TreeSet[(A,Int)]) returns `TreeSet`.
For now I would keep it as is. I see currently no reason to change it.